### ISSUE TITLE
CSV.write - floats

### ISSUE THREAD

**Issue by bkamins** [JuliaData/CSV.jl#492]
When outputting floats when writing a CSV file it would be nice to allow users to control what kind of notation should be used (maybe allow to pass a formatter). The issue is that:
1. sometimes you are OK with the default
2. sometimes you may want to enforce scientific notation
3. sometimes you might want to enforce rounding (this is needed when later you want to read in the file using other software that does not accept that many decimal digits as are normally printed by Julia - what I mean is that eg. `3.141592653589793` is parsed incorrectly by some readers and you need to round it to e.g. 8 decimal digits)

This is a minor issue.

**Comment by quinnj** [comment]
Marking as 1.X milestone, meaning I don't plan on implementing this before 1.0 release (which will happen soon). I was originally hoping https://github.com/JuliaLang/julia/pull/32859 would land in a reasonable amount of time, but alas, it has fallen by the wayside of paltry review. My next solution is to just put the functionality in a package and we can use it here.

**Comment by juliohm** [comment]
@quinnj any update on this feature? We are working on a project where we need to export the floating point numbers without scientific notation. Any option to save all floating point numbers with a fixed number of decimals?

**Comment by quinnj** [comment]
No update as of yet, but I'm hoping to have time to address this soon as we do a blitz on CSV open issues.

**Comment by jkoefinger** [comment]
I agree with @bkamins. Would be great if this could be fixed.  A single notation makes it easier to share data with people using other programming languages.

**Comment by quinnj** [comment]
This was fixed in the 1.0 release by the floatformat keyword ([#1196](https://github.com/JuliaData/CSV.jl/pull/1196)). [issue resolution reviewed by AI; please respond with objections or additional feedback/concerns/issues]
